packages/inter-protocol/src/vpool-xyk-amm/pool.js

```diff
@@ -252,18 +253,21 @@ export const definePoolKind = (baggage, ammPowers, storageNode, marshaller) => {
         ),
       );
 
-      poolSeat.incrementBy(
-        userSeat.decrementBy(harden({ Liquidity: liquidityIn })),
-      );
-      userSeat.incrementBy(
-        poolSeat.decrementBy(
-          harden({
-            Central: centralTokenAmountOut,
-            Secondary: tokenKeywordAmountOut,
-          }),
-        ),
+      atomicTransfer(
+        ammPowers.zcf,
+        harden([
+          [userSeat, poolSeat, { Liquidity: liquidityIn }],
+          [
+            poolSeat,
+            userSeat,
+            {
+              Central: centralTokenAmountOut,
+              Secondary: tokenKeywordAmountOut,
```

Comment:
amounts are different. I'd split into two lines.

Reply:
This isn't a balance issue. Each TransferArgs line is only doing an internally balanced oneway transfer. We're just aggregating multiple specific amounts into an AmountKeywordRecord, which is what it's for. So we can consider these together as a heterogeneous group of assets.

Reply:
Resolving